src/pages/commits/utils.ts

```diff
@@ -73,18 +77,25 @@ const generateBuildVariantOptionsForTaskIconsFromState = (
 
   let shouldShowTaskIcons = true;
   let statusesToShow = [];
-  if (hasTasks) {
-    statusesToShow = filterState.statuses;
-  } else if (hasStatuses) {
-    const onlyHasNonFailingStatuses =
-      arrayIntersection(filterState.statuses, FAILED_STATUSES).length === 0;
-    if (onlyHasNonFailingStatuses) {
-      shouldShowTaskIcons = false;
+
+  // Don't filter icons if "All" view is applied
+  if (filterState.view === ProjectHealthView.Failed) {
```

Comment:
I think this interaction could use some more consistency. The initial goal of this project was to simplify visual pattern matching of task icons on the project health page. The addition of the toggle is a positive step towards achieving that goal, as it allows users to view all of their icons. However, the lack of consistency and the forced grouping when filters are applied creates a suboptimal user experience in my opinion. I'd suggest keeping things consistent no matter if filters are on or off. That way, users can have a smooth experience and easily find what they need.